src/components/tabs/__next__/tabs.component.tsx

```diff
@@ -0,0 +1,417 @@
+import React, { useCallback, useEffect, useState } from "react";
+
+import Typography from "../../typography";
+import { TabsProvider, useTabs } from "./tabs.context";
+import { TabListProps, TabPanelProps, TabProps, TabsProps } from "./tabs.types";
+import {
+  Spacer,
+  StyledScrollButton,
+  StyledScrollButtonPlaceholder,
+  StyledTab,
+  StyledTabList,
+  StyledTabListWrapper,
+  StyledTabs,
+} from "./tabs.style";
+import Logger from "../../../__internal__/utils/logger";
+import useResizeObserver from "../../../hooks/__internal__/useResizeObserver";
+import Icon from "../../icon";
+import { TabProvider } from "./tab.context";
+
+export const TabPanel = ({ children, id, tabId }: TabPanelProps) => {
+  const { activeTab } = useTabs();
+
+  return (
+    <TabProvider tabId={tabId} visible={tabId === activeTab}>
+      <div id={id} role="tabpanel" aria-labelledby={tabId}>
+        {children}
+      </div>
+    </TabProvider>
+  );
+};
+
+let unsupportedSlotConfigurationWarningTriggered = false;
+
+export const Tab = ({
+  controls,
+  error = false,
+  id,
+  label,
+  leftSlot,
+  rightSlot,
+  warning = false,
+  info = false,
+}: TabProps) => {
+  const [internalError, setInternalError] = useState<boolean | string>(error);
+  const [internalWarning, setInternalWarning] = useState<boolean | string>(
+    warning,
+  );
+  const [internalInfo, setInternalInfo] = useState<boolean | string>(info);
+
+  const {
+    activeTab,
+    focusIndex,
+    orientation,
+    setActiveTab,
+    setCurrentTabId,
+    size,
+    errors,
+    warnings,
+    infos,
+  } = useTabs();
+  const selected = activeTab === id;
+
+  useEffect(() => {
+    if (selected) setCurrentTabId(id);
+  }, [id, selected, setCurrentTabId]);
+
+  if (
+    (!!leftSlot || !!rightSlot) &&
+    typeof label !== "string" &&
+    !unsupportedSlotConfigurationWarningTriggered
+  ) {
+    Logger.warn(
+      "[WARNING] Using `leftSlot` and/or `rightSlot` is not supported when `label` is not a string. Please use `leftSlot` and/or `rightSlot` alongside a string `label`, or use the `label` prop exclusively.",
+    );
+    unsupportedSlotConfigurationWarningTriggered = true;
+  }
+
+  useEffect(() => {
+    if (focusIndex === id) {
+      const tabElement = document.getElementById(id);
+      tabElement?.focus();
+    }
+  }, [focusIndex, id, setCurrentTabId]);
+
+  /** Can't be unit-tested; controlled by form tests */
+  /* istanbul ignore next */
+  useEffect(() => {
+    let tabErrorEntries = errors[id];
+    let tabWarningEntries = warnings[id];
+    let tabInfoEntries = infos[id];
+
+    if (error && !tabErrorEntries) {
+      tabErrorEntries = { static: error };
+    }
+
+    if (warning && !tabWarningEntries) {
+      tabWarningEntries = { static: warning };
+    }
+
+    if (info && !tabInfoEntries) {
+      tabInfoEntries = { static: info };
+    }
+
+    if (!tabErrorEntries) {
+      setInternalError(false);
+      return;
+    } else {
```

Comment:
comment: the `else` clauses here seem superfluous as you're `return`ing in the `if`